base/util.jl

```diff
@@ -750,3 +750,47 @@ kwdef_val(::Type{Cwstring}) = Cwstring(C_NULL)
 kwdef_val{T<:Integer}(::Type{T}) = zero(T)
 
 kwdef_val{T}(::Type{T}) = T()
+
+"""
+    @callsuper f(x::T, y::S, ...)
+
+Call the method of function `f` with parameter signature specified by `T`, `S`, etc.
+"""
+macro callsuper(ex)
+    ex.head == :call || error("@invoke requires a call expression")
+    fname = ex.args[1]
+    args = ex.args[2:end]
+    types = Symbol[]
+    vals = Symbol[]
+    kwargs = Any[]
+    blk = quote end
+    for arg in args
+        if isa(arg,Expr) && arg.head == :kw
```

Comment:
This is not entirely correct. A few cases that should be handled and possibly tested are. (These are the reason I started but didn't finish the pr yet...)

1. Evaluation order

    Currently `:parameters` are always evaluated the first. It's not necessarily the best choice but this should be consisent with normal calls.

2. `:parameters` can have `...`

3. `:parameters` can have non `:kw` in it

4. positional arguments can also have `...`

5. if `parameter` only have `...`s and all of them are empty, the non-kw version should be called.